### PR TITLE
Adds support for specifying a list of available drill sizes.

### PR DESCRIPTION
**Pull request by kmjonsson**
New options:

--drills-tolerance -- Acceptable tolerance from available drills (default 0.1mm)
--drills-available -- List of available drills separated width ','.
                        Tolerance can be specified with added '':<-tol>:<+tol>''

This makes it possible to only output drill code with the drill bits available.
It also masks of drill files with small differance of drill bit size.

### CODE REVIEW

**Comment by eyal0** [comment]
Why is this useful?  Couldn't I just use the existing code and then when it asks me for a drill bit, I use whatever it closest in size?

File: options.cpp [hunk 1]
```diff
@@ -248,6 +248,8 @@ options::options()
        ("drill-speed", po::value<int>(), "spindle rpm when drilling")
        ("drill-front", po::value<bool>()->implicit_value(true), "[DEPRECATED, use drill-side instead] drill through the front side of board")
        ("drill-side", po::value<string>()->default_value("auto"), "drill side; valid choices are front, back or auto (default)")
+       ("drills-available", po::value<string>(), "List of available drills separated width ','. Tolerance can be specified with added '':<-tol>:<+tol>''")
```
Comment (eyal0) [inline options.cpp]: The type here should not be string, it should be `vector<available_drillbit>`.  You will need to override `operator>>` and `operator<<` for `vector<available_drillbit>`.  There are examples in units.hpp.  Look for the overrides for `Software`, for example.

**Comment by kmjonsson** [comment]
It's useful in that sense that you can optimize the route of drilling if you have an output with lots of similar drill bit sizes. And the parking and continue time can be quite annoying.

File: options.cpp (hunk 1 above)
Comment (kmjonsson) [reply options.cpp]: OK, I look into that.

**Comment by laurivosandi** [comment]
Thumbs up for this feature!

**Comment by eyal0** [comment]
It's a complicated feature and I'm not sure how to implement it. Specifying
so many parameters like min and max size for each drill seems clunky. What
if there is an overlap? What if a rounding error causes a drill size to be
3.9999 and it's less than 4?

**Comment by kmjonsson** [comment]
@eyal0 For my usage specifying a list of available with a tolerance has worked well. My current implementation adds the possibility to add tolerance for each drill.
The implementation uses the drills in the order put in the argument so if there is an overlap it will choose the first one.

I always look at the output before I put it into the machine to see if there are strange errors (like your suggested rounding error).

I implemented this the first time when I had an gerber with 9(?!?) drillsizes around 1mm. it os 0.975, 0.982, 0.984, ..., 1.09. The time for the machine going up to park and pressing enter and running all around the pcb kind of made me do the work with this.

I'll currently had a ton of work at work so I have had no time thinking of it.

I'll will do it as soon as I have time.

**Comment by eyal0** [comment]
I fixed this in a mostly compatible way.  Please try it out!  Thanks.

**Pull request closed without merging**